ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ProcessesExportImportResourcesSpec.scala

```diff
@@ -95,18 +110,34 @@ class ProcessesExportImportResourcesSpec extends FlatSpec with ScalatestRouteTes
       val modified = processDetails.copy(metaData = processDetails.metaData.copy(id = "SOMEVERYFAKEID"))
 
       val multipartForm =
-        FileUploadUtils.prepareMultiPart(jsonMarshaller.marshallToString(UiProcessMarshaller.toJson(modified)), "process")
+        FileUploadUtils.prepareMultiPart(UiProcessMarshaller.toJson(modified).spaces2, "process")
 
       Post(s"/processes/import/${processToSave.id}", multipartForm) ~> routeWithAllPermissions ~> check {
         status shouldEqual StatusCodes.BadRequest
       }
     }
   }
 
+  test("export pdf") {
+    val processToSave = ProcessTestData.sampleDisplayableProcess
+    saveProcess(processToSave) {
+      status shouldEqual StatusCodes.OK
+
+      val testSvg = "<svg viewBox=\"0 0 120 70\" xmlns=\"http://www.w3.org/2000/svg\">\n  " +
+        "<path d=\"M20,20h20m5,0h20m5,0h20\" stroke=\"#c00000\" stroke-width=\"10\"/>\n  " +
+        "<path d=\"M20,40h20m5,0h20m5,0h20M30,30v20m25,0v-20m25,0v20\" stroke=\"#008000\" stroke-width=\"6\"/>\n</svg>"
+
+      Post(s"/processesExport/pdf/${processToSave.id}/2", HttpEntity(testSvg)) ~> routeWithAllPermissions ~> check {
+        //TODO: check it's pdf?
+        status shouldEqual StatusCodes.OK
+      }
+    }
+
+  }
+
   private def assertProcessPrettyPrinted(response: String, process: CanonicalProcess): Unit = {
-    response shouldBe UiProcessMarshaller.toJson(process).spaces2
-      .replaceAll(Pattern.quote("[]"), "[ ]")
-      .replaceAll("\\{[ \n]+\\}", "{ }")
+    response.replace("\n", "").replace(" ", "") shouldBe
```

Comment:
`response shouldBe UiProcessMarshaller.toJson(process).spaces2`? I think it's good idea to add assertions that response contains new lines.

Reply:
true, I added sth nicer...